packages/components/src/JSONSchemaRenderer/JSONSchemaRenderer.component.js

```diff
@@ -0,0 +1,174 @@
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import entries from 'lodash/entries';
+
+import css from './JSONSchemaRenderer.scss';
+
+const className = 'json-schema-renderer';
+
+/**
+ * UnkownTypeException
+ *
+ * @param type - The unkown type
+ * @returns {Object} An UnkownTypeException
+ */
+function UnkownTypeException(type) {
+	this.name = 'UnkownTypeException';
+	this.message = `Unkown type: ${type}`;
+}
+
+/**
+ * InvalidSchemaException
+ *
+ * @returns {Object} An InvalidSchemaException
+ */
+function InvalidSchemaException() {
+	this.name = 'InvalidSchemaException';
+	this.message = 'Invalid Schema';
+}
+
+/**
+ * textRenderer
+ *
+ * @param key
+ * @param title
+ * @param text
+ * @returns {string} - HTML markup for a text component
+ */
+function textRenderer(key, title, text) {
+	return (
+		<div key={key}>
+			<dt>{title || key}</dt>
+			<dd>{text}</dd>
+		</div>
+	);
+}
+
+/**
+ * arrayRenderer
+ *
+ * @param key
+ * @param title
+ * @param items
+ * @returns {string} - HTML markup for an array component
+ */
+function arrayRenderer(key, title, items) {
+	return (
+		<div className={css.array} key={key}>
+			<dt>{title || key}</dt>
+			{items.map((val, i) => <dd key={`key-${i}`}>{val}</dd>)}
+		</div>
+	);
+}
+
+/**
+ * objectRenderer
+ *
+ * @param key
+ * @param title
+ * @param properties
+ * @param schema
+ * @returns {string} - HTML markup for an object component
+ */
+function objectRenderer(key, title, properties, schema) {
+	const flattenProperties = entries(properties);
+	const elements = flattenProperties.map(typeResolver(schema[key].properties));
+	return (
+		<div className={css.object} key={key}>
+			<h2>{title || key}</h2>
+			<div>
+				{elements}
+			</div>
+		</div>
+	);
+}
+
+const registry = {
+	string: textRenderer,
+	integer: textRenderer,
+	array: arrayRenderer,
+	object: objectRenderer,
+};
+
+/**
+ * typeResolver
+ *
+ * @param schema - The JSONSchema of the data being rendered
+ * @throws {UnkownTypeException} Type must be part of the registry
+ * @returns {Function} resolver receiving data from a map operation
+ */
+function typeResolver(schema) {
+	return function resolver(e) {
+		if (!schema[e[0]]) {
+			return null;
+		}
+		const type = schema[e[0]].type;
+		const title = schema[e[0]].title;
+
+		const renderer = registry[type];
```

Comment:
No not yet. But it could make sense to extend it later. The current registry covers most of the JSONSchema cases.